src/adapter/z-stack/znp/zpiObject.ts

```diff
@@ -14,69 +16,69 @@ const BufferAndListTypes = [
     ParameterType.BUFFER42,
     ParameterType.BUFFER100,
     ParameterType.LIST_UINT16,
-    ParameterType.LIST_ROUTING_TABLE,
-    ParameterType.LIST_BIND_TABLE,
-    ParameterType.LIST_NEIGHBOR_LQI,
     ParameterType.LIST_NETWORK,
-    ParameterType.LIST_ASSOC_DEV,
     ParameterType.LIST_UINT8,
 ];
 
 class ZpiObject {
+    public readonly type: Type;
     public readonly subsystem: Subsystem;
-    public readonly command: string;
-    public readonly commandID: number;
+    public readonly command: MtCmd;
     public readonly payload: ZpiObjectPayload;
-    public readonly type: Type;
-
-    private readonly parameters: MtParameter[];
+    public readonly unpiFrame: UnpiFrame;
 
-    private constructor(type: Type, subsystem: Subsystem, command: string, commandID: number, payload: ZpiObjectPayload, parameters: MtParameter[]) {
+    private constructor(type: Type, subsystem: Subsystem, command: MtCmd, payload: ZpiObjectPayload, unpiFrame: UnpiFrame) {
+        this.type = type;
         this.subsystem = subsystem;
         this.command = command;
-        this.commandID = commandID;
         this.payload = payload;
-        this.type = type;
-        this.parameters = parameters;
+        this.unpiFrame = unpiFrame;
     }
 
     public static createRequest(subsystem: Subsystem, command: string, payload: ZpiObjectPayload): ZpiObject {
         if (!Definition[subsystem]) {
             throw new Error(`Subsystem '${subsystem}' does not exist`);
         }
 
-        const cmd = Definition[subsystem].find((c: MtCmd): boolean => c.name === command);
-        if (cmd?.request === undefined) {
+        const cmd = Definition[subsystem].find((c) => c.name === command);
+        if (cmd === undefined || isMtCmdAreqZdo(cmd) || cmd?.request === undefined) {
             throw new Error(`Command request '${command}' from subsystem '${subsystem}' not found`);
         }
 
-        return new ZpiObject(cmd.type, subsystem, command, cmd.ID, payload, cmd.request);
-    }
+        // Create the UnpiFrame
+        const buffalo = new BuffaloZnp(Buffer.alloc(MaxDataSize));
+        for (const parameter of cmd.request) {
+            const value = payload[parameter.name];
+            buffalo.write(parameter.parameterType, value, {});
+        }
+        const buffer = buffalo.getWritten();
+        const upiFrame = new UnpiFrame(cmd.type, subsystem, cmd.ID, buffer);
```

Comment:
typo in variable name, `unpiFrame` I assume.

Reply:
Fixed!